### PR TITLE
PLT-2155 adding transaction history days

### PR DESCRIPTION
**Pull request by ramesh-token**

### CODE REVIEW

File: user/src/main/java/io/token/user/AccessTokenBuilder.java [hunk 1]
```diff
@@ -117,6 +117,8 @@ public static AccessTokenBuilder fromTokenRequest(TokenRequest tokenRequest) {
         if (tokenRequest.getRequestPayload().hasActingAs()) {
             builder.setActingAs(tokenRequest.getRequestPayload().getActingAs());
         }
+        builder.getAccessBuilder().setTransactionHistoryDays(tokenRequest.getRequestPayload().getAccessBody().getTransactionHistoryDays());
```
Comment (sibinlu) [inline user/src/main/java/io/token/user/AccessTokenBuilder.java:120]: We need to provide a set endpoint for users to set TransactionHistoryDays with the builder. e.g. setRefId.

File: core/src/main/java/io/token/tokenrequest/TokenRequest.java [hunk 2]
```diff
@@ -443,6 +443,18 @@ public TokenRequest build() {
                             .setAccountResourceList(list)
                             .build());
         }
+
+        /**
+         * optional set the transaction history days.
```
Comment (sibinlu) [inline core/src/main/java/io/token/tokenrequest/TokenRequest.java:448]: Please follow the other Javadoc style.

File: core/src/main/java/io/token/tokenrequest/TokenRequest.java [hunk 3]
```diff
@@ -443,6 +443,18 @@ public TokenRequest build() {
...
+         * @param transactionHistoryDays transactionHistoryDays
```
Comment (sibinlu) [inline core/src/main/java/io/token/tokenrequest/TokenRequest.java:449]: here as well.

File: core/src/main/java/io/token/tokenrequest/TokenRequest.java (hunk 2 above)
Comment (ramesh-token) [reply core/src/main/java/io/token/tokenrequest/TokenRequest.java]: Updated java doc.

File: core/src/main/java/io/token/tokenrequest/TokenRequest.java (hunk 3 above)
Comment (ramesh-token) [reply core/src/main/java/io/token/tokenrequest/TokenRequest.java]: updated java doc.

**Review by sibinlu — APPROVED** [review]

**Review by sibinlu — APPROVED** [review]
Reminder: merge it only if your changes are in production to avoid any use from our clients.

**Pull request merged**